eve/service/svc_note.py

```diff
@@ -0,0 +1,172 @@
+                    "AlterationDate": "2021-04-24",
+                    "DueDate": "2021-04-28",
+                    "Priority": 2,
+                    "NoteTxt": "This is sample entry 3."
+                }
+            ]'''
+            self.touch = json.loads(self.touch)
+            with open(self.path, 'w', encoding='utf-8') as f:
+                json.dump(self.touch, f, indent=2, ensure_ascii=False)
+            print(f'Created {self.path}')
+        # Initialize instance variables
+        self.data = None
+        self.new_note = None
+
+    def output(self):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
+
+        # transform null-values to str('N/A')
+        for dic in self.data:
+            for key in dic:
+                if dic[key] is None:
+                    dic[key] = 'N/A'
+
+        # transform all values to str
+        for dic in self.data:
+            for key in dic:
+                dic[key] = str(dic[key])
+
+        # order by priority and due_date
+        self.data = sorted(self.data, key=lambda k: k['Priority'])
+        self.data = sorted(self.data, key=lambda k: k['DueDate'])
+
+        # output data
+        console = Console()
+        table = Table(show_header=True,
+                      box=box.ASCII,
+                      #  show_lines=True,
+                      header_style='bold blue')
+        table.add_column('uid', style='dim')
+        table.add_column('creation date', style='dim')
+        table.add_column('alteration date', style='dim')
+        table.add_column('due date', style='dim')
+        table.add_column('priority', style='dim')
+        table.add_column('note text')
+        for dic in self.data:
+            row = list()
+            for key in dic:
+                row.append(dic[key])
+            table.add_row(str(row[0]), str(row[1]), str(row[2]), str(row[3]), str(row[4]), str(row[5]))
+        console.print(table)
+
+    def add_note(self, note_txt, due_date=None, priority=None):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
+
+        # compose new_note
+        uid = int(len(self.data)+1)
+        self.new_note = dict()
+        self.new_note["UID"] = uid
+        self.new_note["CreationDate"] = str(date.today())
+        self.new_note["AlterationDate"] = None
+        self.new_note["DueDate"] = due_date
+        self.new_note["Priority"] = priority
+        self.new_note["NoteTxt"] = note_txt
+
+        # append new_note to old_notes
+        self.data.append(self.new_note)
+
+        # write to JSON-file
+        with open(self.path, 'w', encoding='utf-8') as f:
+            json.dump(self.data, f, indent=2, ensure_ascii=False)
+
+        print(f'New note with uid {uid} added.')
+
+    def del_note(self, uid):
+        # read JSON-file
+        with open(self.path, encoding='utf-8') as f:
+            self.data = json.load(f)
```

Comment:
You seem to be doing this a lot, it is probably better to read this once in the instantiation of the class

Reply:
Yeah. I forgot, that I could instantiate the data in the __init__. That was because I first modeled the output-function where I had to convert integers to strings and so on. But I will restructure.